core/src/contracts/stablex_contract.rs

```diff
@@ -80,6 +80,14 @@ pub struct FilteredOrderPage {
     pub next_page_user_offset: u16,
 }
 
+#[derive(thiserror::Error, Debug)]
+pub enum NoopTransactionError {
+    #[error("no account")]
+    NoAccount,
+    #[error("execution error")]
+    ExecutedOrder(ExecutionError),
```

Comment:
nit: You can derive the `from` implementation so that you don't have to `map_err`:
```suggestion
    ExecutedOrder(#[from] ExecutionError),
```

Reply:
I added this but still need map_err because we're not using `?`. Makes sense to have it anyway.